src/darwin/list.mm

```diff
@@ -131,6 +131,7 @@ DeviceDescriptor CreateDeviceDescriptorFromDiskDescription(std::string diskBsdNa
       CFRelease(diskDescription);
       CFRelease(disk);
     }
+    [dl dealloc];
```

Comment:
You're not supposed to call `dealloc` - never. The only place where you can do this is inside `dealloc` method where you must call `[super dealloc]` as the last thing. Should be `release`.

Reply:
amended